src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieClusterMonitor.scala

```diff
@@ -95,17 +96,16 @@ class ZombieClusterMonitor(config: ZombieClusterConfig, gdDAO: GoogleDataprocDAO
   }
 
   private def isClusterActiveInGoogle(cluster: Cluster): Future[Boolean] = {
-    // Clusters in Creating status may not yet exist in Google. Therefore treat all Creating clusters as active.
-    if (cluster.status == ClusterStatus.Creating) {
-      Future.successful(true)
-    } else {
-      // Check if status returned by GoogleDataprocDAO is an "active" status.
-      gdDAO.getClusterStatus(cluster.googleProject, cluster.clusterName) map { clusterStatus =>
-        ClusterStatus.activeStatuses contains clusterStatus
-      } recover { case e =>
-        logger.warn(s"Unable to check status of cluster ${cluster.projectNameString} for zombie cluster detection", e)
-        true
-      }
+
+    val secondsSinceClusterCreation: Long = Duration.between(cluster.auditInfo.createdDate, Instant.now()).getSeconds
+    //this or'd with the google cluster status gives creating clusters a grace period before they are marked as zombies
+    val isWithinHangTolerance =  cluster.status == ClusterStatus.Creating && secondsSinceClusterCreation > config.creationHangTolerance.toSeconds
+
+   gdDAO.getClusterStatus(cluster.googleProject, cluster.clusterName) map { clusterStatus =>
+      (ClusterStatus.activeStatuses contains clusterStatus) || isWithinHangTolerance
```

Comment:
I could be misreading, but I think this is not quite right. `ClusterStatus.activeStatuses` includes `Creating`, so we won't actually be checking `isWithinHangTolerance` when status is `Creating`.

Reply:
As I understand it based, if the cluster is active in google, it is not considered a zombie in any scenario, as the point of the zombie monitor is to remove clusters in Leo DB that aren't present in google, and the functionality is not designed to extend the other way.

Thus, the hang tolerance is only relevant in the cases in which the cluster is not active in google. That being said, I think I should split line 102 into two separate variables so it is more clear though (after I merge this maybe, so I don't have to go through the automation tests again). Something like:

`val shouldConsiderCreatingClusterActive = isWithinHangTolerance && cluster.status == ClusterStatus.Creating`

Which should maintain:

| Creating  | In google | Within hang tolerance | Desired Result |
|---|---|---|---|
| Yes | Yes | Yes | true |
| Yes | No | Yes | true |
| Yes | Yes | No | true |
| Yes | No  | No | false |
| No | Yes | Any | true |
| No | No | Any | false |

Reply:
thanks for the explanation. I think that makes sense. one nit, the method name is a bit misleading after we add the hangTolerance logic. `isClusterActive` would be better. A comment like what you said here would be helpful too.